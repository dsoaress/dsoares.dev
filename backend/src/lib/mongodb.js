import { MongoClient } from 'mongodb'

const { MONGODB_URI, MONGODB_DB } = process.env

let cached = global.mongo

if (!cached) {
  cached = global.mongo = { conn: null, promise: null }
}

export async function connectToDatabase() {
  if (cached.conn) {
    return cached.conn
  }

  if (!cached.promise) {
    const opts = {
      useNewUrlParser: true,
      useUnifiedTopology: true
    }

    cached.promise = MongoClient.connect(MONGODB_URI, opts).then(client => {
      return {
        client,
        db: client.db(MONGODB_DB)
      }
    })
  }
  cached.conn = await cached.promise
  return cached.conn
}
