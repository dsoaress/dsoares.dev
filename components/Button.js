import cn from 'classnames'

export default function Button({ children, ...rest }) {
  return (
    <button className={button} {...rest}>
      {children}
    </button>
  )
}

const button = cn(
  'px-6',
  'py-3',
  'font-bold',
  'transition',
  'duration-300',
  'border',
  'rounded-md',
  'text-primary-50',
  'bg-primary-700',
  'hover:bg-primary-800',
  'dark:bg-primary-500',
  'dark:hover:bg-primary-600',
  'text-semibold',
  'border-primary-600',
  'dark:border-primary-400',
  'hover:shadow-lg'
)
