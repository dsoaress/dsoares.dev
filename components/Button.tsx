import { darken, lighten } from 'polished'
import { ButtonHTMLAttributes } from 'react'
import styled from 'styled-components'

const StyledButton = styled.button`
  transition: background 0.3s cubic-bezier(0.4, 0, 0.2, 1);
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 1.5rem;
  height: 2.5rem;
  border: none;
  border-radius: 0.35rem;
  background: ${({ theme }) => theme.colors.primary};
  color: ${({ theme }) => lighten(1, theme.colors.text)};
  font-size: 0.85rem;
  font-weight: 600;
  cursor: pointer;

  &:hover {
    background: ${({ theme }) => darken(0.1, theme.colors.primary)};
  }

  &:active {
    background: ${({ theme }) => darken(0.2, theme.colors.primary)};
  }
`

export default function Button({ ...props }: ButtonHTMLAttributes<HTMLButtonElement>) {
  return <StyledButton {...props} />
}
