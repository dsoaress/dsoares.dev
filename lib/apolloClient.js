import { ApolloClient, InMemoryCache } from '@apollo/client'

const { API_URL } = process.env

const client = new ApolloClient({
  uri: `${API_URL}/graphql`,
  cache: new InMemoryCache()
})

export default client
