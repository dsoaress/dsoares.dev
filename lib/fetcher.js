import { api } from './api'

export const fetcher = (...args) => api.get(...args).then(res => res.data)
