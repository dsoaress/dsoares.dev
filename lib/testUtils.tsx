import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ReactElement } from 'react'
import { ThemeProvider } from 'styled-components'

import { theme } from '@/styles/theme'

const Providers = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'queries'>): RenderResult =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
