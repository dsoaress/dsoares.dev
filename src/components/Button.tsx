import cn from 'classnames'
import type { ButtonHTMLAttributes } from 'react'
import type { IconType } from 'react-icons/lib'

type ButtonProps = {
  icon?: IconType
} & ButtonHTMLAttributes<HTMLButtonElement>

export function Button({ className, children, icon: Icon, ...props }: ButtonProps) {
  return (
    <button
      className={cn(
        'flex h-11 cursor-pointer items-center rounded-md border-0 bg-primary-600 px-6 text-sm font-medium leading-none text-neutral-50 transition-colors duration-300 hover:bg-primary-700 active:bg-primary-800',
        className
      )}
      {...props}
    >
      {!!Icon && <Icon className="mr-2 text-lg" />}
      {children}
    </button>
  )
}
