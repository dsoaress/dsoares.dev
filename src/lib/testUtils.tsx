import { render } from '@testing-library/react'
import { ComponentType, ReactElement, ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'

import { theme } from '@/styles/theme'

type ProvidersProps = {
  children?: ReactNode
}

const Providers = ({ children }: ProvidersProps) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options = {}) =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
