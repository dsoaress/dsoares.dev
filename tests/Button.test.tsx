import Button from '@/components/Button'
import { render, screen } from '@/lib/testUtils'

describe('<Button />', () => {
  it('Renders with children', () => {
    render(<Button>Children</Button>)

    expect(screen.getByText('Children')).toBeInTheDocument()
    expect(screen.getByText('Children')).toMatchSnapshot()
  })
})
