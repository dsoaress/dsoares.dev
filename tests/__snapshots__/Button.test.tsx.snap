// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Renders with children 1`] = `
.c0 {
  -webkit-transition: background 0.3s cubic-bezier(0.4,0,0.2,1);
  transition: background 0.3s cubic-bezier(0.4,0,0.2,1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1.5rem;
  height: 2.5rem;
  border: none;
  border-radius: 0.35rem;
  background: #F43F5E;
  color: #fff;
  font-size: 0.85rem;
  font-weight: 600;
  cursor: pointer;
}

.c0:hover {
  background: #f10f36;
}

.c0:active {
  background: #c20b2a;
}

<button
  class="c0"
>
  Children
</button>
`;
